js/src/views/Dapps/dapps.js

```diff
@@ -111,58 +88,49 @@ export default class Dapps extends Component {
     return (
       <div
         className={ styles.item }
-        key={ app.url }>
+        key={ app.id }>
         <Summary app={ app } />
       </div>
     );
   }
 
-  renderGlobalApps () {
-    const { globalApps } = this.state;
+  loadImages () {
+    const { available } = this.state;
+    const { dappReg } = Contracts.get();
 
-    return globalApps.map(this.renderApp);
+    return Promise.all(available.map((app) => dappReg.getImage(sha3(app.id))))
```

Comment:
The issue is now that we do the expensive sha3 every cycle and as authors have no way of seeing what it is to manage :(